CMakeLists.txt

```diff
@@ -103,7 +85,33 @@ int main() {
   return __builtin_ctzll(0);
 }" HAVE_BUILTIN_CTZ)
 
-check_cxx_source_compiles("
+
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "i686.*|i386.*|x86.*|amd64.*|AMD64.*|x86_64.*")
```

Comment:
only x86_64 and i386 are needed (uname -p)
ref: https://cmake.org/cmake/help/latest/variable/CMAKE_HOST_SYSTEM_PROCESSOR.html#variable:CMAKE_HOST_SYSTEM_PROCESSOR

Reply:
Yes it is probably overkill, and i'm more than happy to change/minimize it.  We had some problems with two builds at one point for a project I worked on (reicast emulator) and did not get the standard results.  

For reference current (7.0) msys2/mingw
MINGW64: ``CMAKE_SYSTEM_PROCESSOR: "AMD64"``
MINGW32: ``CMAKE_SYSTEM_PROCESSOR: "AMD64"``  ( yes this is it's output )

$ rm -rf ./.* ./* ; file /mingw32/bin/gcc.exe ;  uname -a ; cmake.exe -G "MinGW Makefiles" ..
/mingw32/bin/gcc.exe: PE32 executable (console) Intel 80386 (stripped to external PDB), for MS Windows
MINGW32_NT-10.0-19041 dev-station 3.0.7-338.x86_64 2019-07-11 10:58 UTC x86_64 Msys

CMAKE_SYSTEM_PROCESSOR: "AMD64"
CMAKE_SIZEOF_VOID_P: "4"
CMAKE_C_COMPILER_ID: "GNU"


The other system didn't match x86_64 || i386 was a modified unix (bsd) toolchain.   Which brings up the possibility for a cmake toolchain file to specify anything It wants via   set(CMAKE_SYSTEM_PROCESSOR x64)

Anyhow I will limit it to the normal out